coffea/nanoevents/methods/vector.py

```diff
@@ -417,6 +598,75 @@ def delta_r(self, other):
         """
         return numpy.sqrt(self.delta_r2(other))
 
+    @awkward.mixin_class_method(numpy.negative)
+    def negative(self):
+        """Returns the negative of the vector"""
+        return awkward.zip(
+            {
+                "x": - self.x,
+                "y": - self.y,
+                "z": - self.z,
+                "t": - self.t
+            },
+            with_name="LorentzVector",
+        )
+
+    @property
+    def pvec(self):
+        """The `x`, `y` and `z` compontents as a `ThreeVector`"""
+        return awkward.zip(
+            {
+                "x": self.x,
+                "y": self.y,
+                "z": self.z
+            },
+            with_name="ThreeVector"
+        )
+
+    @property
+    def boostvec(self):
+        """The `x`, `y` and `z` compontents divided by `t` as a `ThreeVector`
+
+        This can be used for boosting.
+        """
+        pvec = self.pvec
+        # Allow self to be a zero vector by using awkward.where
+        mask = self.t == 0
+        t = awkward.where(mask, 1, self.t)
+        out = awkward.where(
+            mask,
+            awkward.zeros_like(pvec),
+            self.pvec / t,
+            highlevel=False
+        )
+        return awkward.Array(out, behavior=self.behavior)
+
+    def boost(self, other):
+        """Apply a Lorentz boost given by the `ThreeVector` `other` and return it
+
+        Note that this follows the convention that, for example in order to boost
+        a vector into its own rest frame, one needs to use the negative of its `boostvec`
+        """
+        b2 = other.rho2
+        gamma = (1 - b2) ** (-0.5)
```

Comment:
We should guard somehow against the case `b2 == 1` as well as `b2 == 0`.

Reply:
Right now for `b2==1` there will be a warning and `inf` as vector components. Is there anything better result in this situation?

Reply:
That seems reasonable, I wasn't sure what the behavior was currently.